src/DSPC.sol

```diff
@@ -0,0 +1,290 @@
+    /// @dev Sets wards[usr] to 1 and emits Rely event
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /// @notice Revoke authorization from an address
+    /// @param usr The address to be deauthorized
+    /// @dev Sets wards[usr] to 0 and emits Deny event
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /// @notice Add a facilitator
+    /// @param usr The address to add as a facilitator
+    /// @dev Sets buds[usr] to 1 and emits Kiss event. Facilitators can propose rate updates but cannot modify system parameters
+    function kiss(address usr) external auth {
+        buds[usr] = 1;
+        emit Kiss(usr);
+    }
+
+    /// @notice Remove a facilitator
+    /// @param usr The address to remove as a facilitator
+    /// @dev Sets buds[usr] to 0 and emits Diss event
+    function diss(address usr) external auth {
+        buds[usr] = 0;
+        emit Diss(usr);
+    }
+
+    /// @notice Configure module parameters
+    /// @param what The parameter to configure (only "bad" is supported)
+    /// @param data The value to set (must be 0 or 1 for "bad")
+    /// @dev Emits File event after successful configuration
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "bad") {
+            require(data <= 1, "DSPC/invalid-bad-value");
+            bad = data;
+        } else {
+            revert("DSPC/file-unrecognized-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /// @notice Configure constraints for a rate
+    /// @param id The rate identifier (ilk name, "DSR", or "SSR")
+    /// @param what The parameter to configure ("min", "max", or "step")
+    /// @param data The value to set (must be greater than 0)
+    /// @dev Emits File event after successful configuration
+    function file(bytes32 id, bytes32 what, uint256 data) external auth {
+        if (what == "min") {
+            _cfgs[id].min = uint16(data);
+        } else if (what == "max") {
+            require(data > 0, "DSPC/invalid-max");
+            _cfgs[id].max = uint16(data);
+        } else if (what == "step") {
+            require(data > 0, "DSPC/invalid-step");
+            _cfgs[id].step = uint16(data);
+        } else {
+            revert("DSPC/file-unrecognized-param");
+        }
+
+        emit File(id, what, data);
+    }
+
+    /// @notice Apply rate updates
+    /// @param updates Array of rate updates to apply
+    /// @dev Each update is validated against configured constraints before being applied
+    /// @dev Emits Put event after all updates are successfully applied
+    /// @dev Reverts if:
+    ///      - Empty updates array
+    ///      - Rate below configured minimum
+    ///      - Rate above configured maximum
+    ///      - Rate change exceeds configured step size
+    function put(ParamChange[] calldata updates) external toll good {
```

Comment:
I thought we agreed to rename this method to `set` instead of `put`.
Please let me know if that was not the case.
```suggestion
    function set(ParamChange[] calldata updates) external toll good {
```